core/src/main/java/hudson/model/Run.java

```diff
@@ -1429,7 +1429,9 @@ public String toString() {
     /**
      * Returns the log file.
      * @return The file may reference both uncompressed or compressed logs
-     */  
+     * @deprecated Assumes file-based storage of the log. Use other methods giving various kinds of streams.
```

Comment:
nit: maybe tweak slightly to mention JEP-210 and list some of the recommended APIs:
```suggestion
     * @deprecated Assumes file-based storage of the log, which is not necessarily the case for Pipelines after JEP-210. Use other methods giving various kinds of streams such as {@link Run#getLogReader()},  {@link Run#getLogInputStream()}, or {@link Run#getLogText()}.
```